pkg/render/addon.go

```diff
@@ -67,15 +91,71 @@ func Addon(templateSource, destPath string) error {
 	return nil
 }
 
-func renderTemplate(content []byte) ([]byte, error) {
-	result := bytes.NewBufferString("")
-	tmpl, err := template.New("").Parse(string(content))
+func renderTemplate(tmpl *template.Template, versionFilePath string) ([]byte, error) {
+	envMap, err := generate_version_info_map(versionFilePath)
 	if err != nil {
 		return nil, err
 	}
-	err = tmpl.Execute(result, nil)
+
+	result := bytes.NewBufferString("")
+	err = tmpl.Execute(result, envMap)
 	if err != nil {
 		return nil, err
 	}
 	return result.Bytes(), nil
 }
+
+// generate_version_info_map reads the version_info file in project root
+// and renders a raw template which can be used for subsequent processing
+func generate_version_info_map(versionFilePath string) (map[string]string, error) {
+	versionFile, err := os.Open(versionFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("error opening version file: %v", err)
+	}
+	defer versionFile.Close()
+	versionScanner := bufio.NewScanner(versionFile)
+
+	var lines []string
+	for versionScanner.Scan() {
+		lines = append(lines, versionScanner.Text())
+	}
+
+	result := make(map[string]string, len(lines))
+	for _, v := range lines {
+		fields := strings.Split(v, "=")
+		if len(fields) == 2 { // need check to ignore hash=bang directive in version info file
+			if strings.Contains(fields[0], imageSuffix) {
```

Comment:
This way, the `- image: rancher/harvester-eventrouter:v0.2.0` will be a bit tricky

or we just define the right infomation in the version_info
e.g.:
HARVESTER_SEEDER_IMAGE_TAG="v0.3"
HARVESTER_EVENTROUTER_IMAGE="rancher/harvester-eventrouter:v0.2.0"

The render just replace the var with env, no special processing about the `imageSuffix`.

Anohter solution is just to skip the `IMAGE` suffix for `harvester-eventrouter`

Reply:
seems only the second solution to skip `image` suffix, the `HARVESTER_SEEDER_IMAGE` is directly used in build-bundle

```
  # seeder images
  echo ${HARVESTER_SEEDER_IMAGE} >> ${image_list_file}
  ```

Reply:
we can update build-bundle as well to bring it in sync with what the addon definition will be, there is no hard rule that the variable name cannot be changed 😄